internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -115,48 +104,19 @@ func (a *Adapter) EnsureSnapshotExists() (result controller.OperationResult, err
 		return controller.RequeueWithError(err)
 	}
 	if len(*existingSnapshots) > 0 {
-		if len(*existingSnapshots) == 1 {
-			existingSnapshot := (*existingSnapshots)[0]
-			a.logger.Info("There is an existing Snapshot associated with this build pipelineRun, but the pipelineRun is not yet annotated",
-				"snapshot.Name", existingSnapshot.Name)
-			err := a.annotateBuildPipelineRunWithSnapshot(&existingSnapshot)
-			if err != nil {
-				a.logger.Error(err, "Failed to update the build pipelineRun with snapshot name",
-					"pipelineRun.Name", a.pipelineRun.Name)
-				return controller.RequeueWithError(err)
```

Comment:
Since there might be return `RequeueWithError(err)` here, I think we may need to replace `controller.ContinueProcessing()` below with the returned `result` and `error`, HDYT?

Reply:
Good idea, added.